### PR TITLE
[Drop-In UI] Moved RecenterButtonComponent outside Drop-In UI

### PR DESCRIPTION
**Pull request by tomaszrybakiewicz**
Part of https://github.com/mapbox/navigation-sdks/issues/1780

### TL;DR

This PR adds new ComponentInstaller for a component that re-centers the map to device location on button click.

```kotlin
MapboxNavigationApp.installComponents(/* LifecycleOwner */ this) {
    recenterButton(mapView, recenterButton)
}
```

### Description
- Moved `RecenterButtonComponent` from `libnavui-dropin` -> `libnavui-maps`
- Introduced `RecenterButtonComponentContract` that defines `RecenterButtonComponent` controller logic.
- Added default `RecenterButtonComponentContract` implementation (`MapboxRecenterButtonComponentContract`) that when attached to MapboxNavigation instance, registers itself as `LocationObserver` and recenters the map to last know location on button click.
- Introduced `ComponentInstaller.recenterButton(..)`
- Introduced `ComponentInstaller.components(...)` that allows installation of multiple components at once.
- Updated `MapboxNavigationObserverChain.removeAndDetach(...)` to allow removal and detaching of multiple components.
- Added new ComponentInstaller API example to qa-test-app.

### CODE REVIEW

**Review by abhishek1508 — COMMENTED** [review]
In the example activities, we are showing recenter button always, but it doesn't work unless we start navigation. We should hide the recenter button and make it visible only when the navigation starts.

https://user-images.githubusercontent.com/9770186/184803824-b05ab062-07b2-489a-947b-a7a87c608f12.mp4

**Comment by tomaszrybakiewicz** [comment]
> In the example activities, we are showing recenter button always, but it doesn't work unless we start navigation. We should hide the recenter button and make it visible only when the navigation starts.

Are you referring to this particular QA App example or `MapboxRecenterButtonComponentContract` implementation?
Since applicaiton creates and styles the button, shouldn't it be responsible for its visibility?

**Comment by abhishek1508** [comment]
> > In the example activities, we are showing recenter button always, but it doesn't work unless we start navigation. We should hide the recenter button and make it visible only when the navigation starts.
> 
> Are you referring to this particular QA App example or `MapboxRecenterButtonComponentContract` implementation?
> Since applicaiton creates and styles the button, shouldn't it be responsible for its visibility?
> 

I am referring to the 2 example activities that are modified in this PR. It's kind of annoying that the recenter button is there outside active navigation and it doesn't work. We should hide the button when the activity is started and show it when click on startNavigation

File: qa-test-app/src/main/res/layout/components_activity_layout.xml [hunk 1]
```diff
@@ -29,14 +29,30 @@
         android:text="Start Navigation"
         android:background="@color/primary" />
 
-    <com.mapbox.navigation.ui.voice.view.MapboxAudioGuidanceButton
-        android:id="@+id/soundButton"
+    <androidx.appcompat.widget.LinearLayoutCompat
```
Comment (abhishek1508) [inline qa-test-app/src/main/res/layout/components_activity_layout.xml:32]: Looking at this change I understand that the plan is to keep adding more buttons to this linear layout as they come in correct? For ex: the next button will be `CameraModeButton`

@tomaszrybakiewicz

File: qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/ComponentsAltActivity.kt [hunk 2]
```diff
@@ -84,8 +87,18 @@ class ComponentsAltActivity : AppCompatActivity() {
...
+                    .zoom(10.0)
+                    .bearing(0.0)
+                    .build()
+                animationOptions = MapAnimationOptions.Builder()
+                    .duration(1000L)
+                    .build()
+            }
 
             // custom components
+            component(MapButtonsVisibility(binding.mapButtons))
```
Comment (abhishek1508) [inline qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/ComponentsAltActivity.kt:101]: If my understanding https://github.com/mapbox/mapbox-navigation-android/pull/6158#discussion_r947357188 is correct, then this logic might not work. When we add `CameraModeButton` in the next PR, it will also hide/show based on `TripSessionState`, because camera mode button is a part of the same linear layout `binding.mapButtons`. However the camera mode button should always be visible no matter in whatever state you are in.

I think we should control the visibility of these buttons individually and not as a group because not all the buttons have the same requirement of when should they hide or show.

@tomaszrybakiewicz

**Comment by abhishek1508** [comment]
Looking at this further

```
MapboxNavigationApp.installComponents(this) {
    audioGuidanceButton(binding.soundButton)
    routeLine(binding.mapView)
    routeArrow(binding.mapView)
    recenterButton(binding.mapView, binding.recenterButton)

    // custom components
    component(DropInStartReplayButton(binding.startNavigation))
```

the problem is that we are starting trip session only when clicked on start navigation. Since trip session doesn't start when activity is started, clicking recenter doesn't work. While hiding recenter initially and showing only on active navigation works, this is kind of a hack. The correct fix should actually modify the code above to start the trip session right away when the activity starts. 

This is out of the scope of this PR and should be followed up in a different one. 
@tomaszrybakiewicz let's revert the [last commit](https://github.com/mapbox/mapbox-navigation-android/pull/6158/commits/78c3ddf3879a2258c234a89158e4eac06b59295f) as this is not the correct solution for the problem I mentioned [here](https://github.com/mapbox/mapbox-navigation-android/pull/6158#issuecomment-1216938791). Once the last commit is reverted, I will ✅ this PR as rest everything else looks good.

File: qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/ComponentsAltActivity.kt (hunk 2 above)
Comment (abhishek1508) [reply qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/ComponentsAltActivity.kt:101]: Resolving this based on https://github.com/mapbox/mapbox-navigation-android/pull/6158#issuecomment-1217357778

File: qa-test-app/src/main/res/layout/components_activity_layout.xml (hunk 1 above)
Comment (abhishek1508) [reply qa-test-app/src/main/res/layout/components_activity_layout.xml:32]: Resolving based on https://github.com/mapbox/mapbox-navigation-android/pull/6158#issuecomment-1217357778

**Comment by tomaszrybakiewicz** [comment]
> This is out of the scope of this PR and should be followed up in a different one.
> @tomaszrybakiewicz let's revert the [last commit](https://github.com/mapbox/mapbox-navigation-android/pull/6158/commits/78c3ddf3879a2258c234a89158e4eac06b59295f) as this is not the correct solution for the problem I mentioned [here](https://github.com/mapbox/mapbox-navigation-android/pull/6158#issuecomment-1216938791). Once the last commit is reverted, I will ✅ this PR as rest everything else looks good.

Done

**Review by abhishek1508 — APPROVED** [review]
Thanks for addressing the feedback.
